### PR TITLE
feat/AB#54824_grouping-layers

### PR DESCRIPTION
**Pull request by matheus-relief**
# Description

This PR makes it possible to group map layer on the map settings.

## Ticket
https://dev.azure.com/WHOHQ/EMSSAFE/_boards/board/t/App%20Builder%20-%20WHE%20Dashboard/Stories?System.AssignedTo=%40me

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
See screenshots

## Sreenshots
Peek 2023-05-29 13-51

# Checklist:

( * == Mandatory ) 

- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/safe/src/i18n/en.json [hunk 1]
```diff
@@ -562,15 +557,6 @@
 			"subject": "Subject",
 			"withoutRecords": "Without records"
 		},
-		"emailPreview": {
```
Comment (AntoineRelief) [inline libs/safe/src/i18n/en.json:565]: @matheus-relief 
why did you remove that?

File: libs/safe/src/lib/components/widgets/map-settings/map-forms.ts [hunk 2]
```diff
@@ -63,13 +63,17 @@ const DEFAULT_GRADIENT = [
  */
 export const createLayerForm = (value?: LayerModel) => {
   const type = get(value, 'type', 'FeatureLayer') as LayerType;
-  return fb.group({
+
+  return new FormGroup({
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/widgets/map-settings/map-forms.ts:67]: @matheus-relief 
please use form builder !

Comment (AntoineRelief) [reply libs/safe/src/lib/components/widgets/map-settings/map-forms.ts:67]: I'll merge it like that, but please explain me why you did this change

File: libs/safe/src/i18n/en.json (hunk 1 above)
Comment (matheus-relief) [reply libs/safe/src/i18n/en.json:565]: Not on purpose, happened when I pulled from origin a while back and there were no translations for the other languages, so I just went with it, I can put it back and add the french keys if you want.

File: libs/safe/src/lib/components/widgets/map-settings/map-forms.ts (hunk 2 above)
Comment (matheus-relief) [reply libs/safe/src/lib/components/widgets/map-settings/map-forms.ts:67]: Is there a reason why we should prioritize the form builder? 

But yeah, there's a difference in behavior (or maybe I just don't know how to use `fb` properly). I was having an issue with the sublayers. For some reason, when I used the fb, and there wasn't any sublayers when creating the form, the value for the sublayers was initialized as `null` instead of an empty array. I tried a bunch of things, including `fb.array`, but nothing really worked. So I just rewrote it using this syntax (that I'm more familiar with) and it solved the issue

Comment (matheus-relief) [reply libs/safe/src/lib/components/widgets/map-settings/map-forms.ts:67]: btw, I was using the same `get` function, with the `[]` as fallback, and still got `null`.
